app/code/Magento/Ui/view/base/web/js/lib/core/collection.js

```diff
@@ -224,7 +224,7 @@ define([
         regionHasElements: function (name) {
             var region = this.getRegion(name);
 
-            return region().length > 0;
+            return region().length > 0 && 'item' in region.first();
```

Comment:
return !_.isUndefined(region()[0]) && region()[0].elems.length > 0;

Reply:
@mrtuvn It doesn't work well for payment methods. I also can't provide jasmine test for my fix as i always got some issues in testing observable array.